tidy3d/plugins/adjoint/components/medium.py

```diff
@@ -75,25 +63,67 @@ def _get_volume_disc(
 
         return vol_coords, d_vol
 
-    # pylint:disable=too-many-arguments
-    def field_contribution(
+    # pylint: disable=too-many-arguments
+    def e_mult_volume(
         self,
         field: Literal["Ex", "Ey", "Ez"],
         grad_data_fwd: FieldData,
         grad_data_adj: FieldData,
+        vol_coords: Dict[str, np.ndarray],
+        d_vol: float,
+    ) -> xr.DataArray:
+        """Get the E_fwd * E_adj * dV field distribution inside of the discretized volume."""
+
+        e_fwd = grad_data_fwd.field_components[field]
+        e_adj = grad_data_adj.field_components[field]
+        e_dotted = e_fwd * e_adj
+        e_dotted = d_vol * e_dotted.real.interp(**vol_coords).isel(f=0).values
+        return xr.DataArray(e_dotted, coords=vol_coords)
```

Comment:
Why do `.values` on the previous line and then convert to `DataArray` again, doesn't that make it the same thing?

Reply:
yea I think this is just a thing where I hadn't cleaned this up yet. I just replaced
```py
e_dotted = d_vol * e_dotted.real.interp(**vol_coords).isel(f=0).values
return xr.DataArray(e_dotted, coords=vol_coords)
```
with
```py
return d_vol * e_dotted.real.interp(**vol_coords).isel(f=0)
```
and it seems fine